gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceCallableFactoryClassComposer.java

```diff
@@ -60,8 +61,17 @@ protected TransportContext getTransportContext() {
     return transportContext;
   }
 
+  protected Transport getTransport() {
```

Comment:
Noticed that we have a `getTransportContext` methods for most of the abstract composers, can we leverage `getTransportContext().transport()` for our use case instead of adding new methods?

Reply:
I was thinking of using this, except it doesn't work for grpc+rest workflows. When I call `getTransportContext().transport()`, it would return `Transport.GRPC_REST`. If I pass that to the sub-composers, the sub-composers wouldn't know if it was REST or GRPC.

Reply:
I would consider `Transport.GRPC_REST` the same as `Transport.REST` when we are checking if there is any REST supported methods, as that's practically what we are doing right now. Otherwise having two methods `getTransportContext()` and `getTransport()` serve similar purposes are confusing.

Reply:
I agree that for this feat that `Transport.GRPC_REST the same as Transport.REST`. The issue I came across was that there is a bunch of logic in the AbstractComposer classes that shouldn't generate if is REST transport. But both sub-composers have a TransportContext of Transport.GRPC_REST. Since both sub-composers technically have the same Transport, it's impossible to build logic where `GRPC_REST == REST` in the Abstract class.

I was trying to find a generic solution that wouldn't require me to hard code `Transport.REST` : `service.hasAnyEnabledMethodsForTransport(Transport.REST)`

I was thinking that `service.hasAnyEnabledMethodsForTransport(getTransport())` would be clearer for the Abstract Classes as it shouldn't be tied to any transport, but if it's confusing to add a similar method then I can just hardcode the REST transport.